sync/awsdatabricks.py

```diff
@@ -981,6 +999,18 @@ def _get_run_cluster_id(tasks: List[dict]) -> Response[str]:
         return Response(error=DatabricksError(message="More than 1 cluster found for tasks"))
 
 
+def _get_run_spark_context_id(tasks: List[dict]) -> Response[str]:
+    context_ids = {task["cluster_instance"]["spark_context_id"] for task in tasks}
+    num_ids = len(context_ids)
+
+    if num_ids == 1:
+        return Response(result=context_ids.pop())
+    elif num_ids == 0:
+        return Response(error=DatabricksError(message="No cluster found for tasks"))
+    else:
+        return Response(error=DatabricksError(message="More than 1 cluster found for tasks"))
```

Comment:
The `spark_context_id` for a given run is now passed as an argument to `_get_eventlog`. The `spark_context_id` helps us identify, in the event of a shared/re-used cluster, which directory actually contains the eventlog we're interested in. The S3 logic hasn't been updated to use this value yet, but I found it useful for the DBFS directory traversal